app/scenes/Document/components/Document.tsx

```diff
@@ -614,7 +614,7 @@ class DocumentScene extends React.Component<Props> {
                 </Flex>
               </React.Suspense>
             </MaxWidth>
-            {isShare && !isCustomDomain() && (
+            {isShare && !parseDomain(window.location.origin).custom && (
```

Comment:
I changed back to having a single interface for all the domain parsing stuff -- this mimics the native `URL` class. The call sites that previously referenced any of the exported functions now all just call `parseDomain` directly and use the properties of the return object